File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        
        Map<Integer,Integer> mp=new HashMap<>();

        int n=nums.length;
        List<Integer> freq[] =new List[n+1];
        for(int i=0;i<n+1;i++)
        {
            freq[i]=new ArrayList<>();
        }
        for(int i :nums)
        {
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        for(Map.Entry<Integer,Integer> entry : mp.entrySet())
        {
            freq[entry.getValue()].add(entry.getKey());
        }
        int res[] =new int[k];
        int index=0;
        for(int i=freq.length-1;i>0 && index<k;i--)
        {
            for(int num :freq[i])
            {
                res[index++]=num;
                if(index==k) return res;
            }

        }
        return res;


    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:

        res = [num for num in nums[:k]]
        heapq.heapify(res)

        for num in nums[k:]:

            if num > res[0]:
                heapq.heappop(res)
                heapq.heappush(res,num)
            
        
        return res[0]
        